Lib/ast.py

```diff
@@ -305,28 +306,13 @@ def get_docstring(node, clean=True):
     return text
 
 
-def _splitlines_no_ff(source):
-    """Split a string into lines ignoring form feed and other chars.
-
-    This mimics how the Python parser splits source code.
-    """
-    idx = 0
+_line_pattern = re.compile(r"(.*?(?:\r\n|\n|\r|$))")
```

Comment:
Since this is changing the behavior in a way that is not so obvious, I'd really love to see some test cases (on the `get_source_segment` side, not this internal function). By looking at the #103285, I see that you already tested this on a various pieces of source code, I think it is a nice idea to port them (and maybe add a couple more) so that we can ensure this functionality performs as intended.

Reply:
There are already plenty of `get_source_segment()` tests in `test_ast.py` (all the tests with `_check_content`). Is there anything specific you want to test?

Reply:
There are already tests for single-line statements ending with each newline. Also segments from a single line and multiple lines. I added a test set specifically targeting different new line characters. One for a function with different newlines(the function source would include the newline) which would fail with my original incorrect implementation. I also added a mixed line source which would also fail.